src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs

```diff
@@ -121,7 +121,8 @@ private static bool IsInvalidGrant(string errorCode, string subErrorCode)
 
         private static bool IsInvalidGrantSubError(string subError)
         {
-            if (string.IsNullOrEmpty(subError))
+            if (string.IsNullOrEmpty(subError)
+                || subError.Contains("50196"))
```

Comment:
not sure exactly how this resolves the issue. also, use a constant for the value here. Do we have any tests to update/add?

Reply:
Used const and added a test.